xmtp_mls/migrations/2023-10-29-205333_state_machine_init/up.sql

```diff
@@ -0,0 +1,72 @@
+CREATE TABLE groups (
+    -- Random ID generated by group creator
+    "id" BLOB PRIMARY KEY NOT NULL,
+    -- Based on the timestamp of the welcome message
+    "created_at_ns" BIGINT NOT NULL,
+    -- Enum of GROUP_MEMBERSHIP_STATE
+    "membership_state" INT NOT NULL
```

Comment:
could we use TEXT as the enum instead of an integer? 

Something like 

` membership_state      TEXT CHECK( membership_state IN ('allowed','rejected','pending') )`

Reply:
I'm easy there. A few bytes of extra state isn't going to kill us. It would require a database migration every time we add an allowed enum value to update that check, as opposed to the int where it's just a code change. But it's more readable in the DB which is worth something.

Reply:
that's true, I didn't consider the migration aspect. Maybe we just leave it as int for now. I'm not strongly for either way I was just writing up the rust part and had the idea